### PR TITLE
Copy Prune Information when cloning Traversal Nodes

### PR DESCRIPTION
**Pull request by maierlars**
### Scope & Purpose

OneShard setup transforms a traversal node into a `LocalTraversalNode`. During that process information required for pruning were not copied. Thus no pruning was done. Additionally: https://github.com/arangodb/enterprise/pull/585

- [x] :hankey: Bugfix (requires CHANGELOG entry)
- [ ] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification
- [ ] :book: CHANGELOG entry made

#### Backports:

- [ ] No backports required
- [x] Backports required for: 3.7

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket number:
- [ ] Design document: 

### Testing & Verification

*(Please pick either of the following options)*

- [ ] This change is a trivial rework / code cleanup without any test coverage.
- [x] The behavior in this PR was *manually tested*
- [ ] This change is already covered by existing tests, such as *(please describe tests)*.
- [ ] This PR adds tests that were used to verify all changes:
  - [ ] Added new C++ **Unit tests**
  - [x] Added new **integration tests** (e.g. in shell_server / shell_server_aql)
  - [ ] Added new **resilience tests** (only if the feature is impacted by failovers)
- [x] There are tests in an external testing repository: https://github.com/arangodb/enterprise/pull/585
- [ ] I ensured this code runs with ASan / TSan or other static verification tools

Link to Jenkins PR run:

### Documentation

> All new features should be accompanied by corresponding documentation. 
> Bugs and features should furthermore be documented in the CHANGELOG so that
> support, end users, and other developers have a concise overview. 

- [ ] Added entry to *Release Notes* 
- [ ] Added a new section in the *Manual* 
- [ ] Added a new section in the *HTTP API* 
- [ ] Added *Swagger examples* for the HTTP API  
- [ ] Updated license information in *LICENSES-OTHER-COMPONENTS.md* for 3rd party libraries

### External contributors / CLA Note 

Please note that for legal reasons we require you to sign the [Contributor Agreement](https://www.arangodb.com/documents/cla.pdf)
before we can accept your pull requests.

### CODE REVIEW

**Comment by maierlars** [comment]
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/12719/

**Comment by maierlars** [comment]
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/12721/

**Review by goedderz — APPROVED** [review]
LGTM

Maybe you also want to rename `usesVertexOutVariable()` to `vertexOutVariableIsUsedLater()`, and similarly for the other two.

**Comment by maierlars** [comment]
image

**Comment by maierlars** [comment]
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/12726/

File: CHANGELOG [hunk 1]
```diff
@@ -1,6 +1,8 @@
 devel
 -----
 
+* Fixed a bug that ignored prune statements in the one shard setup.
```
Comment (goedderz) [inline CHANGELOG:4]:
```suggestion
* Fixed an AQL bug that ignored PRUNE statements in OneShard setups.
```

**Comment by maierlars** [comment]
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/12933/

**Comment by maierlars** [comment]
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/12935/

**Comment by maierlars** [comment]
image

**Pull request merged**